File: delivery-service/db/src/main/java/org/delivery/db/userOrder/UserOrderRepository.java
package org.delivery.db.userOrder;

/*public interface UserOrderRepository extends JpaRepository<UserOrderEntity, Long> {
    
    //특정유저의 모든 주문
    //select * from user_order where user_id = ? and status = "REGISTERED" order by id desc
    List<UserOrderEntity> findAllByUserIdAndStatusOrderByIdDesc(Long userId, UserOrderStatus status);

    //select * from user_order where user_id = ? and status in(?)order by id desc
    List<UserOrderEntity> findAllByUserIdAndStatusInOrderByIdDesc(Long userId, List<UserOrderStatus> status);
    
    //특정주문
    //select * from user_order where id = ? and status = "REGISTERED" and user_id = ?
    Optional<UserOrderEntity> findAllByIdAndStatusAndUserId(Long id, UserOrderStatus status, Long userId);

    Optional<UserOrderEntity> findAllByIdAndUserId(Long id, Long userId);
}*/

File: delivery-service/db/src/main/kotlin/org/delivery/db/userOrder/UserOrderRepository.kt
package org.delivery.db.userOrder

import org.delivery.db.userOrder.enums.UserOrderStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderRepository : JpaRepository<UserOrderEntity, Long> {
    //특정유저의 모든 주문
    //select * from user_order where user_id = ? and status = "REGISTERED" order by id desc
    fun findAllByUserIdAndStatusOrderByIdDesc(userId: Long?, status: UserOrderStatus?): List<UserOrderEntity>

    //select * from user_order where user_id = ? and status in(?)order by id desc
    fun findAllByUserIdAndStatusInOrderByIdDesc(userId: Long?, status: List<UserOrderStatus>?): List<UserOrderEntity>

    //특정주문
    //select * from user_order where id = ? and status = "REGISTERED" and user_id = ?
    fun findAllByIdAndStatusAndUserId(id: Long?, status: UserOrderStatus?, userId: Long?): UserOrderEntity?
    fun findAllByIdAndUserId(id: Long?, userId: Long?): UserOrderEntity?
}
